version: '3'
services:
 redis-server:
  image: 'redis:6.2-alpine'
  environment:
   - REDIS_HOST=redis-server
   - REDIS_PORT=6379
 web:
  build: .
  ports:
    - "80:80"
  volumes:
    - ./main.py:/app/main.py
    - ./puppycompanyblog/:/app/puppycompanyblog
  environment:
     POSTGRES_USER: postgres
     POSTGRES_PASSWORD: asengame
     POSTGRES_HOST: db
     POSTGRES_PORT: 5432
     POSTGRES_DB: postgres_db
  tty: true
  restart: on-failure
 tests:
  build: .
  volumes:
    - ./main.py:/app/main.py
    - ./puppycompanyblog/:/app/puppycompanyblog
  command: ["pytest", "./puppycompanyblog/test.py"]
 db:
  image: 'postgres:12.7-alpine'
  environment:
    POSTGRES_DB: postgres_db
    POSTGRES_USER: postgres
    POSTGRES_PASSWORD: asengame
  ports:
    - "5432:5432"
  volumes:
    - /var/lib/psql/data:/var/lib/psql/data 
